rest/blip_sync.go

```diff
@@ -1235,21 +1240,32 @@ func (ctx *blipSyncContext) setActiveSubChanges(changesActive bool) {
 
 // setUseDeltas will set useDeltas on the blipSyncContext as long as both sides of the connection have it enabled.
 func (ctx *blipSyncContext) setUseDeltas(clientCanUseDeltas bool) {
-	// Both sides want deltas
-	if ctx.sgCanUseDeltas && clientCanUseDeltas {
-		if !ctx.useDeltas {
-			base.DebugfCtx(ctx.blipContextDb.Ctx, base.KeySync, "Enabling deltas for this replication")
-			ctx.dbStats.StatsDeltaSync().Add(base.StatKeyDeltaPullReplicationCount, 1)
-			ctx.useDeltas = true
-		}
+	if ctx.useDeltas && ctx.sgCanUseDeltas && clientCanUseDeltas {
+		// fast-path for deltas that are already enabled and still wanted on both sides.
+		return
+	}
+
+	// Both sides want deltas, and we've not previously enabled them.
+	if ctx.sgCanUseDeltas && clientCanUseDeltas && !ctx.useDeltas {
+		base.DebugfCtx(ctx.blipContextDb.Ctx, base.KeySync, "Enabling deltas for this replication")
+		ctx.dbStats.StatsDeltaSync().Add(base.StatKeyDeltaPullReplicationCount, 1)
+		ctx.useDeltas = true
 		return
 	}
 
-	// Log when the client doesn't want deltas, but we do
-	if ctx.sgCanUseDeltas && !clientCanUseDeltas {
-		base.InfofCtx(ctx.blipContextDb.Ctx, base.KeySync, "Disabling deltas for this replication based on client setting.")
+	// We don't want deltas, but we'd previously enabled them.
+	if !ctx.sgCanUseDeltas && ctx.useDeltas {
```

Comment:
setUseDeltas is only called in the context of a client's changes response, and sgCanUseDeltas is only set on the initial blip handshake, so this scenario shouldn't be hit today.  Is this just for potential future scenarios where deltas are disabled on the SG side midway through a  replication?

Reply:
That's true. Today replications are only turning deltas on from an off state, or keeping it in an off state.

I've added another fast-path condition to this function to help with the 2 common cases, and included more detail in the test/benchmark.